irods/query.py

```diff
@@ -65,7 +67,16 @@ def add_keyword(self, keyword, value = ''):
 
     def filter(self, *criteria):
         new_q = self._clone()
-        new_q.criteria += list(criteria)
+
+        if self.case_sensitive:
+            new_q.criteria += list(criteria)
+        else:
+            for criterion in criteria:
+                if type(criterion.unprocessed_value) == str:
+                    criterion.unprocessed_value = str.upper(
+                        criterion.unprocessed_value)
```

Comment:
I think this is okay, but I am a little confused about the use of the term `unprocessed_value` when in fact it is being "processed" in a sense and then assigned back to `unprocessed_value`. This might just be me ;)

Maybe `raw_value`? Hmm...

Reply:
I too am confused, but more because unprocessed value (as a `@property` with getter and setter) seems to amount to a synonym for the _value member.  I'd like to see a comment maybe, to explain what added benefit there is to having the property as opposed to just manipulating `self._value`

Reply:
I thought it was best for other classes to not change `_value` directly, considering that it is an internal member by name convention. If I would have renamed `_value` to `value`, the current `value` getter would also have to be renamed. 

I have added a short comment for now. If renaming `_value` is better, what would be a good new name for the current `value` getter? Something like `irods_value` or `quoted_value` perhaps?

Reply:
I'm not the most familiar with the conventions here, but my understanding was that it's best to avoid direct access of the "private" members of Python classes if it can be helped, hence the getter/setter.

We could, alternatively, provide a method to the `criterion` type which performs this action internally. e.g.
```py
def make_case_insensitive(self):
```

I'm not sure that I have a strong view either way

Reply:
> I thought it was best for other classes to not change `_value` directly, considering that it is an internal member by name convention. If I would have renamed `_value` to `value`, the current `value` getter would also have to be renamed.
> 
> I have added a short comment for now. If renaming `_value` is better, what would be a good new name for the current `value` getter? Something like `irods_value` or `quoted_value` perhaps?

Yes, perhaps `quoted_value`.  I'm liberal-minded to whatever is chosen, since @stsnel is probably the person with the best understanding of this code at the moment.